### PR TITLE
[jit] Extract the intrinsics generation functions into a separate intrinsics.c file.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
This reduces the size of method-to-ir.c by about 1.2k lines.

File: mono/mini/mini.h [hunk 1]
```diff
@@ -317,6 +319,18 @@ enum {
 
 #endif
 
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+#define EMIT_NEW_X86_LEA(cfg,dest,sr1,sr2,shift,imm) do { \
```
Comment (jaykrell) [inline mono/mini/mini.h]: spaces after commas?

File: mono/mini/mini.h [hunk 2]
```diff
@@ -2094,6 +2108,10 @@ void              mono_local_regalloc (MonoCompile *cfg, MonoBasicBlock *bb);
 MonoInst         *mono_branch_optimize_exception_target (MonoCompile *cfg, MonoBasicBlock *bb, const char * exname);
 void              mono_remove_critical_edges (MonoCompile *cfg);
 gboolean          mono_is_regsize_var (MonoType *t);
+int               mini_class_check_context_used (MonoCompile *cfg, MonoClass *klass);
+int               mini_method_check_context_used (MonoCompile *cfg, MonoMethod *method);
```
Comment (jaykrell) [inline mono/mini/mini.h]: function names at start of line good for search, despite surrounding inferior convention

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build OS X x64 iOS SDK

**Comment by vargaz** [comment]
@monojenkins build OS X x64 iOS SDK

**Comment by vargaz** [comment]
@monojenkins build Linux x64 Coop Suspend

**Pull request merged**